### DISCUSSION TITLE
Reject a Pull Request during merge

### DISCUSSION THREAD

**Discussion by mattstam** [community/community#34219] [Actions]
## Problem

We have a use-case which requires a Pull Request merge to be rejected *after* a user with write access has sent the request to merge. Basically, *after* they have clicked "Confirm squash & merge":

![Screenshot 2022-09-26 072646](https://user-images.githubusercontent.com/15695189/192303815-c9dfd5f6-d73d-4690-8d94-d021a76524d8.png)

but *before* the commit is added to whichever branch it is trying to merge into.

## What we've tried

Currently we attempt this via this hook point:
```yaml
on:
  pull_request_target:
    types:
      - closed
```
and the action *does* run at the same time the merge happens. We would like this action failing to reject the Pull Request from being merged, but currently, the merge still happens immediately even with a corresponding Action failure.

For more context on the use-case, the a full example is [here](https://github.com/aagusuab/testActionRepo/blob/main/.github/workflows/main.yml). We want to check that the commit message for that Pull Request contains a certain string, and if it does not, reject the merge from happening.

The problem with the typical approach of adding this action as a status check and protecting the branch with `Require status checks to pass before merging` setting, is that the message for the actual PR merge is written *after* the status check workflows run.

It would be great to have another hook point in-between the request to merge from the PR, and the actual merge into the branch, such that it could be rejected. Ideally this would leave the PR in an "Open" state.

**Comment by antonblr** [comment]
Couple question / ideas:
1. can the check that generates / updates the PR message set status to failed and block the PR right away? 
2. could you trigger another job on PR message update (push) event and set the PR status?

**Comment by Blacksmoke16** [comment]
I'd like to bump this feature request for a similar use case. I have a many-repo setup that when a PR is merged it'll sync changes from the monorepo out to the individual repos. Normally when this happens you'd get a commit message in the affected individual repos like `Add some feature (#1234)` where `#1234` is the PR number in the monorepo.

What I'd like to do is enforce that the squash merged commit summary includes the `org/monorepo` prefix, e.g. `org/repo#1234` such that it'll properly link back to the PR in the monorepo versus trying to link to a non-existent PR in each individual repo.

The only way I can think to do this is by having some CI job that could validate/reject the summary. I'm also unable to automate merging because I like to customize the commit description vs just letting it default to all the commit summaries/descriptions related to that PR.
